{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "n_K2ZyB6x8-8"
   },
   "source": [
    "# Setup\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "id": "w6qsW0VZykSD"
   },
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'google.colab'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "\u001b[1;32mC:\\WINDOWS\\TEMP/ipykernel_55728/734207778.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0msys\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 2\u001b[1;33m \u001b[1;32mfrom\u001b[0m \u001b[0mgoogle\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcolab\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mdrive\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfiles\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mModuleNotFoundError\u001b[0m: No module named 'google.colab'"
     ]
    }
   ],
   "source": [
    "import sys\n",
    "from google.colab import drive, files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "id": "XIkP6u05_zwh"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple\r\n",
      "Collecting git+https://github.com/lucasb-eyer/pydensecrf.git\r\n",
      "  Cloning https://github.com/lucasb-eyer/pydensecrf.git to c:\\windows\\temp\\pip-req-build-lrtz1yk9\r\n",
      "  Resolved https://github.com/lucasb-eyer/pydensecrf.git to commit 0d53acbcf5123d4c88040fe68fbb9805fc5b2fb9\r\n",
      "  Installing build dependencies: started\r\n",
      "  Installing build dependencies: finished with status 'done'\r\n",
      "  Getting requirements to build wheel: started\r\n",
      "  Getting requirements to build wheel: finished with status 'done'\r\n",
      "    Preparing wheel metadata: started\r\n",
      "    Preparing wheel metadata: finished with status 'done'\r\n",
      "Building wheels for collected packages: pydensecrf\r\n",
      "  Building wheel for pydensecrf (PEP 517): started\r\n",
      "  Building wheel for pydensecrf (PEP 517): finished with status 'error'\r\n",
      "Failed to build pydensecrf\r\n",
      "Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple\r\n",
      "Requirement already satisfied: setuptools in d:\\anaconda3\\lib\\site-packages (58.0.4)\r\n",
      "Collecting setuptools\r\n",
      "  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/41/82/7f54bbfe5c247a8c9f78d8d1d7c051847bcb78843c397b866dba335c1e88/setuptools-65.5.0-py3-none-any.whl (1.2 MB)\r\n",
      "Installing collected packages: setuptools\r\n",
      "  Attempting uninstall: setuptools\r\n",
      "    Found existing installation: setuptools 58.0.4\r\n",
      "    Uninstalling setuptools-58.0.4:\r\n",
      "      Successfully uninstalled setuptools-58.0.4\r\n",
      "Successfully installed setuptools-65.5.0\r\n",
      "Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple\r\n",
      "Requirement already satisfied: kaggle in d:\\anaconda3\\lib\\site-packages (1.5.12)\r\n",
      "Requirement already satisfied: certifi in d:\\anaconda3\\lib\\site-packages (from kaggle) (2021.10.8)\r\n",
      "Requirement already satisfied: six>=1.10 in d:\\anaconda3\\lib\\site-packages (from kaggle) (1.16.0)\r\n",
      "Requirement already satisfied: urllib3 in d:\\anaconda3\\lib\\site-packages (from kaggle) (1.26.12)\r\n",
      "Requirement already satisfied: tqdm in d:\\anaconda3\\lib\\site-packages (from kaggle) (4.64.1)\r\n",
      "Requirement already satisfied: requests in d:\\anaconda3\\lib\\site-packages (from kaggle) (2.25.1)\r\n",
      "Requirement already satisfied: python-dateutil in d:\\anaconda3\\lib\\site-packages (from kaggle) (2.8.2)\r\n",
      "Requirement already satisfied: python-slugify in d:\\anaconda3\\lib\\site-packages (from kaggle) (5.0.2)\r\n",
      "Requirement already satisfied: text-unidecode>=1.3 in d:\\anaconda3\\lib\\site-packages (from python-slugify->kaggle) (1.3)\r\n",
      "Requirement already satisfied: idna<3,>=2.5 in d:\\anaconda3\\lib\\site-packages (from requests->kaggle) (2.10)\r\n",
      "Requirement already satisfied: chardet<5,>=3.0.2 in d:\\anaconda3\\lib\\site-packages (from requests->kaggle) (4.0.0)\r\n",
      "Requirement already satisfied: colorama in d:\\anaconda3\\lib\\site-packages (from tqdm->kaggle) (0.4.4)\r\n",
      "Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple\r\n",
      "Requirement already satisfied: gdown in d:\\anaconda3\\lib\\site-packages (4.5.3)\r\n",
      "Requirement already satisfied: tqdm in d:\\anaconda3\\lib\\site-packages (from gdown) (4.64.1)\r\n",
      "Requirement already satisfied: filelock in d:\\anaconda3\\lib\\site-packages (from gdown) (3.3.1)\r\n",
      "Requirement already satisfied: beautifulsoup4 in d:\\anaconda3\\lib\\site-packages (from gdown) (4.10.0)\r\n",
      "Requirement already satisfied: requests[socks] in d:\\anaconda3\\lib\\site-packages (from gdown) (2.25.1)\r\n",
      "Requirement already satisfied: six in d:\\anaconda3\\lib\\site-packages (from gdown) (1.16.0)\r\n",
      "Requirement already satisfied: soupsieve>1.2 in d:\\anaconda3\\lib\\site-packages (from beautifulsoup4->gdown) (2.2.1)\r\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in d:\\anaconda3\\lib\\site-packages (from requests[socks]->gdown) (1.26.12)\r\n",
      "Requirement already satisfied: certifi>=2017.4.17 in d:\\anaconda3\\lib\\site-packages (from requests[socks]->gdown) (2021.10.8)\r\n",
      "Requirement already satisfied: idna<3,>=2.5 in d:\\anaconda3\\lib\\site-packages (from requests[socks]->gdown) (2.10)\r\n",
      "Requirement already satisfied: chardet<5,>=3.0.2 in d:\\anaconda3\\lib\\site-packages (from requests[socks]->gdown) (4.0.0)\r\n",
      "Requirement already satisfied: PySocks!=1.5.7,>=1.5.6 in d:\\anaconda3\\lib\\site-packages (from requests[socks]->gdown) (1.7.1)\r\n",
      "Requirement already satisfied: colorama in d:\\anaconda3\\lib\\site-packages (from tqdm->gdown) (0.4.4)\r\n",
      "Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple\r\n",
      "Collecting pytorch-lightning\r\n",
      "  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/00/eb/3b2152f9c3a50d265f3e75529254228ace8a86e9a4397f3004f1e3be7825/pytorch_lightning-1.7.7-py3-none-any.whl (708 kB)\r\n",
      "Requirement already satisfied: tqdm>=4.57.0 in d:\\anaconda3\\lib\\site-packages (from pytorch-lightning) (4.64.1)\r\n",
      "Requirement already satisfied: numpy>=1.17.2 in d:\\anaconda3\\lib\\site-packages (from pytorch-lightning) (1.20.1)\r\n",
      "Requirement already satisfied: torch>=1.9.* in d:\\anaconda3\\lib\\site-packages (from pytorch-lightning) (1.11.0+cu113)\r\n",
      "Requirement already satisfied: tensorboard>=2.9.1 in d:\\anaconda3\\lib\\site-packages (from pytorch-lightning) (2.9.1)\r\n",
      "Requirement already satisfied: pyDeprecate>=0.3.1 in d:\\anaconda3\\lib\\site-packages (from pytorch-lightning) (0.3.2)\r\n",
      "Requirement already satisfied: typing-extensions>=4.0.0 in d:\\anaconda3\\lib\\site-packages (from pytorch-lightning) (4.3.0)\r\n",
      "Requirement already satisfied: torchmetrics>=0.7.0 in d:\\anaconda3\\lib\\site-packages (from pytorch-lightning) (0.9.3)\r\n",
      "Requirement already satisfied: PyYAML>=5.4 in d:\\anaconda3\\lib\\site-packages (from pytorch-lightning) (6.0)\r\n",
      "Requirement already satisfied: packaging>=17.0 in d:\\anaconda3\\lib\\site-packages (from pytorch-lightning) (21.0)\r\n",
      "Requirement already satisfied: fsspec[http]!=2021.06.0,>=2021.05.0 in d:\\anaconda3\\lib\\site-packages (from pytorch-lightning) (2021.10.1)\r\n",
      "Requirement already satisfied: requests in d:\\anaconda3\\lib\\site-packages (from fsspec[http]!=2021.06.0,>=2021.05.0->pytorch-lightning) (2.25.1)\r\n",
      "Requirement already satisfied: aiohttp in d:\\anaconda3\\lib\\site-packages (from fsspec[http]!=2021.06.0,>=2021.05.0->pytorch-lightning) (3.8.3)\r\n",
      "Requirement already satisfied: pyparsing>=2.0.2 in d:\\anaconda3\\lib\\site-packages (from packaging>=17.0->pytorch-lightning) (3.0.4)\r\n",
      "Requirement already satisfied: grpcio>=1.24.3 in d:\\anaconda3\\lib\\site-packages (from tensorboard>=2.9.1->pytorch-lightning) (1.47.0)\r\n",
      "Requirement already satisfied: google-auth-oauthlib<0.5,>=0.4.1 in d:\\anaconda3\\lib\\site-packages (from tensorboard>=2.9.1->pytorch-lightning) (0.4.6)\r\n",
      "Requirement already satisfied: tensorboard-plugin-wit>=1.6.0 in d:\\anaconda3\\lib\\site-packages (from tensorboard>=2.9.1->pytorch-lightning) (1.8.1)\r\n",
      "Requirement already satisfied: werkzeug>=1.0.1 in d:\\anaconda3\\lib\\site-packages (from tensorboard>=2.9.1->pytorch-lightning) (2.0.2)\r\n",
      "Requirement already satisfied: protobuf<3.20,>=3.9.2 in d:\\anaconda3\\lib\\site-packages (from tensorboard>=2.9.1->pytorch-lightning) (3.19.4)\r\n",
      "Requirement already satisfied: google-auth<3,>=1.6.3 in d:\\anaconda3\\lib\\site-packages (from tensorboard>=2.9.1->pytorch-lightning) (2.9.1)\r\n",
      "Requirement already satisfied: markdown>=2.6.8 in d:\\anaconda3\\lib\\site-packages (from tensorboard>=2.9.1->pytorch-lightning) (3.4.1)\r\n",
      "Requirement already satisfied: wheel>=0.26 in d:\\anaconda3\\lib\\site-packages (from tensorboard>=2.9.1->pytorch-lightning) (0.37.0)\r\n",
      "Requirement already satisfied: setuptools>=41.0.0 in d:\\anaconda3\\lib\\site-packages (from tensorboard>=2.9.1->pytorch-lightning) (65.5.0)\r\n",
      "Requirement already satisfied: absl-py>=0.4 in d:\\anaconda3\\lib\\site-packages (from tensorboard>=2.9.1->pytorch-lightning) (1.2.0)\r\n",
      "Requirement already satisfied: tensorboard-data-server<0.7.0,>=0.6.0 in d:\\anaconda3\\lib\\site-packages (from tensorboard>=2.9.1->pytorch-lightning) (0.6.1)\r\n",
      "Requirement already satisfied: six>=1.9.0 in d:\\anaconda3\\lib\\site-packages (from google-auth<3,>=1.6.3->tensorboard>=2.9.1->pytorch-lightning) (1.16.0)\r\n",
      "Requirement already satisfied: rsa<5,>=3.1.4 in d:\\anaconda3\\lib\\site-packages (from google-auth<3,>=1.6.3->tensorboard>=2.9.1->pytorch-lightning) (4.9)\r\n",
      "Requirement already satisfied: cachetools<6.0,>=2.0.0 in d:\\anaconda3\\lib\\site-packages (from google-auth<3,>=1.6.3->tensorboard>=2.9.1->pytorch-lightning) (5.2.0)\r\n",
      "Requirement already satisfied: pyasn1-modules>=0.2.1 in d:\\anaconda3\\lib\\site-packages (from google-auth<3,>=1.6.3->tensorboard>=2.9.1->pytorch-lightning) (0.2.8)\r\n",
      "Requirement already satisfied: requests-oauthlib>=0.7.0 in d:\\anaconda3\\lib\\site-packages (from google-auth-oauthlib<0.5,>=0.4.1->tensorboard>=2.9.1->pytorch-lightning) (1.3.1)\r\n",
      "Requirement already satisfied: importlib-metadata>=4.4 in d:\\anaconda3\\lib\\site-packages (from markdown>=2.6.8->tensorboard>=2.9.1->pytorch-lightning) (4.8.1)\r\n",
      "Requirement already satisfied: zipp>=0.5 in d:\\anaconda3\\lib\\site-packages (from importlib-metadata>=4.4->markdown>=2.6.8->tensorboard>=2.9.1->pytorch-lightning) (3.6.0)\r\n",
      "Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in d:\\anaconda3\\lib\\site-packages (from pyasn1-modules>=0.2.1->google-auth<3,>=1.6.3->tensorboard>=2.9.1->pytorch-lightning) (0.4.8)\r\n",
      "Requirement already satisfied: idna<3,>=2.5 in d:\\anaconda3\\lib\\site-packages (from requests->fsspec[http]!=2021.06.0,>=2021.05.0->pytorch-lightning) (2.10)\r\n",
      "Requirement already satisfied: chardet<5,>=3.0.2 in d:\\anaconda3\\lib\\site-packages (from requests->fsspec[http]!=2021.06.0,>=2021.05.0->pytorch-lightning) (4.0.0)\r\n",
      "Requirement already satisfied: certifi>=2017.4.17 in d:\\anaconda3\\lib\\site-packages (from requests->fsspec[http]!=2021.06.0,>=2021.05.0->pytorch-lightning) (2021.10.8)\r\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in d:\\anaconda3\\lib\\site-packages (from requests->fsspec[http]!=2021.06.0,>=2021.05.0->pytorch-lightning) (1.26.12)\r\n",
      "Requirement already satisfied: oauthlib>=3.0.0 in d:\\anaconda3\\lib\\site-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<0.5,>=0.4.1->tensorboard>=2.9.1->pytorch-lightning) (3.2.0)\r\n",
      "Requirement already satisfied: colorama in d:\\anaconda3\\lib\\site-packages (from tqdm>=4.57.0->pytorch-lightning) (0.4.4)\r\n",
      "Requirement already satisfied: multidict<7.0,>=4.5 in d:\\anaconda3\\lib\\site-packages (from aiohttp->fsspec[http]!=2021.06.0,>=2021.05.0->pytorch-lightning) (6.0.2)\r\n",
      "Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in d:\\anaconda3\\lib\\site-packages (from aiohttp->fsspec[http]!=2021.06.0,>=2021.05.0->pytorch-lightning) (4.0.2)\r\n",
      "Requirement already satisfied: aiosignal>=1.1.2 in d:\\anaconda3\\lib\\site-packages (from aiohttp->fsspec[http]!=2021.06.0,>=2021.05.0->pytorch-lightning) (1.2.0)\r\n",
      "Requirement already satisfied: yarl<2.0,>=1.0 in d:\\anaconda3\\lib\\site-packages (from aiohttp->fsspec[http]!=2021.06.0,>=2021.05.0->pytorch-lightning) (1.8.1)\r\n",
      "Requirement already satisfied: frozenlist>=1.1.1 in d:\\anaconda3\\lib\\site-packages (from aiohttp->fsspec[http]!=2021.06.0,>=2021.05.0->pytorch-lightning) (1.3.1)\r\n",
      "Requirement already satisfied: attrs>=17.3.0 in d:\\anaconda3\\lib\\site-packages (from aiohttp->fsspec[http]!=2021.06.0,>=2021.05.0->pytorch-lightning) (21.2.0)\r\n",
      "Requirement already satisfied: charset-normalizer<3.0,>=2.0 in d:\\anaconda3\\lib\\site-packages (from aiohttp->fsspec[http]!=2021.06.0,>=2021.05.0->pytorch-lightning) (2.0.4)\r\n",
      "Installing collected packages: pytorch-lightning\r\n",
      "Successfully installed pytorch-lightning-1.7.7\r\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  Running command git clone -q https://github.com/lucasb-eyer/pydensecrf.git 'C:\\Windows\\Temp\\pip-req-build-lrtz1yk9'\r\n",
      "  ERROR: Command errored out with exit status 1:\r\n",
      "   command: 'D:\\anaconda3\\python.exe' 'D:\\anaconda3\\lib\\site-packages\\pip\\_vendor\\pep517\\in_process\\_in_process.py' build_wheel 'C:\\WINDOWS\\TEMP\\tmppob_er60'\r\n",
      "       cwd: C:\\Windows\\Temp\\pip-req-build-lrtz1yk9\r\n",
      "  Complete output (24 lines):\r\n",
      "  running bdist_wheel\r\n",
      "  running build\r\n",
      "  running build_py\r\n",
      "  creating build\r\n",
      "  creating build\\lib.win-amd64-cpython-39\r\n",
      "  creating build\\lib.win-amd64-cpython-39\\pydensecrf\r\n",
      "  copying pydensecrf\\utils.py -> build\\lib.win-amd64-cpython-39\\pydensecrf\r\n",
      "  copying pydensecrf\\__init__.py -> build\\lib.win-amd64-cpython-39\\pydensecrf\r\n",
      "  running build_ext\r\n",
      "  building 'pydensecrf.eigen' extension\r\n",
      "  creating build\\temp.win-amd64-cpython-39\r\n",
      "  creating build\\temp.win-amd64-cpython-39\\Release\r\n",
      "  creating build\\temp.win-amd64-cpython-39\\Release\\pydensecrf\r\n",
      "  D:\\VC2015_re\\VC\\BIN\\x86_amd64\\cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -Ipydensecrf/densecrf/include -Ipydensecrf -ID:\\anaconda3\\include -ID:\\anaconda3\\Include -ID:\\VC2015_re\\VC\\INCLUDE -ID:\\VC2015_re\\VC\\ATLMFC\\INCLUDE \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.17763.0\\ucrt\" \"-IC:\\Program Files (x86)\\Windows Kits\\NETFXSDK\\4.6.1\\include\\um\" \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.17763.0\\shared\" \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.17763.0\\um\" \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.17763.0\\winrt\" /EHsc /Tppydensecrf/eigen.cpp /Fobuild\\temp.win-amd64-cpython-39\\Release\\pydensecrf/eigen.obj\r\n",
      "  eigen.cpp\r\n",
      "  D:\\VC2015_re\\VC\\BIN\\x86_amd64\\cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -Ipydensecrf/densecrf/include -Ipydensecrf -ID:\\anaconda3\\include -ID:\\anaconda3\\Include -ID:\\VC2015_re\\VC\\INCLUDE -ID:\\VC2015_re\\VC\\ATLMFC\\INCLUDE \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.17763.0\\ucrt\" \"-IC:\\Program Files (x86)\\Windows Kits\\NETFXSDK\\4.6.1\\include\\um\" \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.17763.0\\shared\" \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.17763.0\\um\" \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.17763.0\\winrt\" /EHsc /Tppydensecrf/eigen_impl.cpp /Fobuild\\temp.win-amd64-cpython-39\\Release\\pydensecrf/eigen_impl.obj\r\n",
      "  eigen_impl.cpp\r\n",
      "  D:\\VC2015_re\\VC\\BIN\\x86_amd64\\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:D:\\anaconda3\\libs /LIBPATH:D:\\anaconda3 /LIBPATH:D:\\anaconda3\\PCbuild\\amd64 /LIBPATH:D:\\VC2015_re\\VC\\LIB\\amd64 /LIBPATH:D:\\VC2015_re\\VC\\ATLMFC\\LIB\\amd64 \"/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\lib\\10.0.17763.0\\ucrt\\x64\" \"/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\NETFXSDK\\4.6.1\\lib\\um\\x64\" \"/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\lib\\10.0.17763.0\\um\\x64\" /EXPORT:PyInit_eigen build\\temp.win-amd64-cpython-39\\Release\\pydensecrf/eigen.obj build\\temp.win-amd64-cpython-39\\Release\\pydensecrf/eigen_impl.obj /OUT:build\\lib.win-amd64-cpython-39\\pydensecrf\\eigen.cp39-win_amd64.pyd /IMPLIB:build\\temp.win-amd64-cpython-39\\Release\\pydensecrf\\eigen.cp39-win_amd64.lib\r\n",
      "  eigen.obj : warning LNK4197: 多次指定导出“PyInit_eigen”；使用第一个规范\r\n",
      "    正在创建库 build\\temp.win-amd64-cpython-39\\Release\\pydensecrf\\eigen.cp39-win_amd64.lib 和对象 build\\temp.win-amd64-cpython-39\\Release\\pydensecrf\\eigen.cp39-win_amd64.exp\r\n",
      "  正在生成代码\r\n",
      "  已完成代码的生成\r\n",
      "  LINK : fatal error LNK1158: 无法运行“rc.exe”\r\n",
      "  error: command 'D:\\\\VC2015_re\\\\VC\\\\BIN\\\\x86_amd64\\\\link.exe' failed with exit code 1158\r\n",
      "  ----------------------------------------\r\n",
      "  ERROR: Failed building wheel for pydensecrf\r\n",
      "ERROR: Could not build wheels for pydensecrf which use PEP 517 and cannot be installed directly\r\n",
      "ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\r\n",
      "spyder 5.1.5 requires pyqt5<5.13, which is not installed.\r\n",
      "spyder 5.1.5 requires pyqtwebengine<5.13, which is not installed.\r\n",
      "conda-repo-cli 1.0.4 requires pathlib, which is not installed.\r\n",
      "anaconda-project 0.10.1 requires ruamel-yaml, which is not installed.\r\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "pip install git+https://github.com/lucasb-eyer/pydensecrf.git\n",
    "pip install --upgrade setuptools\n",
    "pip install kaggle\n",
    "pip install gdown\n",
    "pip install pytorch-lightning"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "id": "tgy-P8-wuZQN"
   },
   "outputs": [],
   "source": [
    "# upload kaggle API key\n",
    "with open(\"kaggle.json\", \"w+\") as f:\n",
    "    f.write('{\"username\":\"giorgio123213\",\"key\":[KEY]}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "id": "tyfqsoYtueJb"
   },
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (TEMP/ipykernel_55728/4173360334.py, line 3)",
     "output_type": "error",
     "traceback": [
      "\u001b[1;36m  File \u001b[1;32m\"C:\\WINDOWS\\TEMP/ipykernel_55728/4173360334.py\"\u001b[1;36m, line \u001b[1;32m3\u001b[0m\n\u001b[1;33m    mkdir ~/.kaggle\u001b[0m\n\u001b[1;37m          ^\u001b[0m\n\u001b[1;31mSyntaxError\u001b[0m\u001b[1;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "# download DSTL data\n",
    "%%bash\n",
    "mkdir ~/.kaggle\n",
    "cp kaggle.json ~/.kaggle/\n",
    "chmod 600 ~/.kaggle/kaggle.json\n",
    "kaggle competitions download -c 'dstl-satellite-imagery-feature-detection' -p ../../ -f sixteen_band.zip\n",
    "kaggle competitions download -c 'dstl-satellite-imagery-feature-detection' -p ../../ -f train_wkt_v4.csv.zip\n",
    "kaggle competitions download -c 'dstl-satellite-imagery-feature-detection' -p ../../ -f grid_sizes.csv.zip\n",
    "unzip -q ../../grid_sizes.csv.zip -d /content/\n",
    "unzip -q ../../train_wkt_v4.csv.zip -d /content/\n",
    "unzip -q ../../sixteen_band.zip -d /content/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "xRQVXUaAyyfk"
   },
   "outputs": [],
   "source": [
    "!git clone https://github.com/rwightman/pytorch-image-models.git"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "dNnF8LLQ6bl7"
   },
   "outputs": [],
   "source": [
    "# hTorch library\n",
    "%%bash\n",
    "git clone https://giorgiozannini:[PASSWORD]@github.com/ispamm/hTorch.git\n",
    "cd hTorch\n",
    "python setup.py install\n",
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "OcucZ5RJ6rp_"
   },
   "outputs": [],
   "source": [
    "sys.path.append('hTorch/')\n",
    "sys.path.append('pytorch-image-models/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "cN9YZb2B62ge"
   },
   "source": [
    "# Hyperparameters\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "RGm1wCyn60No"
   },
   "outputs": [],
   "source": [
    "# hyperparameters settings\n",
    "!cat hTorch/qpsp/constants.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "MrDPQ9CPNFF1"
   },
   "outputs": [],
   "source": [
    "!pip install wandb"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "2h2EZNhrAE25"
   },
   "source": [
    "# Run"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "UWmKFWF6AEYk"
   },
   "outputs": [],
   "source": [
    "import pytorch_lightning as pl\n",
    "from experiments.dstl_dataset import LitDSTL\n",
    "from experiments.models.qswin import SwinTransformer\n",
    "from pytorch_lightning.loggers import WandbLogger\n",
    "wandb_logger = WandbLogger()\n",
    "\n",
    "model = SwinTransformer().cuda()\n",
    "dm = LitDSTL()\n",
    "trainer = pl.Trainer(gpus=1, logger=wandb_logger)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "RlhJ5VFBI35_"
   },
   "outputs": [],
   "source": [
    "trainer.fit(model, dm)"
   ]
  }
 ],
 "metadata": {
  "accelerator": "GPU",
  "colab": {
   "collapsed_sections": [],
   "name": "QPSP.ipynb",
   "provenance": [],
   "toc_visible": true
  },
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
